lib/rules/disallow-curly-braces.js

```diff
@@ -0,0 +1,134 @@
+/**
+ * Disallows curly braces after statements.
+ *
+ * Type: `Array` or `Boolean`
+ *
+ * Values: Array of quoted keywords or `true` to disallow curly braces after the following keywords:
+ *
+ * #### Example
+ *
+ * ```js
+ * "disallowCurlyBraces": [
+ *     "if",
+ *     "else",
+ *     "while",
+ *     "for",
+ *     "do",
+ *     "with"
+ * ]
+ * ```
+ *
+ * ##### Valid
+ *
+ * ```js
+ * if (x) x++;
+ * ```
+ *
+ * ##### Invalid
+ *
+ * ```js
+ * if (x) {
+ *     x++;
+ * }
+ * ```
+ */
+
+var assert = require('assert');
+var defaultKeywords = require('../utils').curlyBracedKeywords;
+
+module.exports = function() {};
+
+module.exports.prototype = {
+
+    configure: function(statementTypes) {
+        assert(
+            Array.isArray(statementTypes) || statementTypes === true,
+            this.getOptionName() + ' option requires array or true value'
+        );
+
+        if (statementTypes === true) {
+            statementTypes = defaultKeywords;
+        }
+
+        this._typeIndex = {};
+        statementTypes.forEach(function(type) {
+            this._typeIndex[type] = true;
+        }.bind(this));
+    },
+
+    getOptionName: function() {
+        return 'disallowCurlyBraces';
+    },
+
+    check: function(file, errors) {
+
+        function isBlockStatement(node) {
+            return node && node.type === 'BlockStatement';
+        }
+
+        function isMultiline(node) {
```

Comment:
This is still wrong, as it'll miss:

```
if (a) {
  a = [
    'b'
  ];
}
```

Instead, count the number of statements inside the block statement, and error if it's 1.